include/dxc/DxilRootSignature/DxilRootSignature.h

```diff
@@ -388,6 +385,30 @@ bool VerifyRootSignature(_In_ const DxilVersionedRootSignatureDesc *pDesc,
                          _In_ llvm::raw_ostream &DiagStream,
                          _In_ bool bAllowReservedRegisterSpace);
 
+class ScopedVersionedRootSignature {
+  const DxilVersionedRootSignatureDesc *m_pRootSignature;
+
+public:
+  ScopedVersionedRootSignature() : m_pRootSignature(nullptr) {}
+  ScopedVersionedRootSignature(
+      const DxilVersionedRootSignatureDesc *pRootSignature)
+  : m_pRootSignature(pRootSignature){}
+  ~ScopedVersionedRootSignature() { 
+      DeleteRootSignature(m_pRootSignature);
+  }
+  const DxilVersionedRootSignatureDesc* operator -> () const {
+    return m_pRootSignature;
+  }
+  const DxilVersionedRootSignatureDesc ** get_address_of() {
+    return &m_pRootSignature;
+  }
+  const DxilVersionedRootSignatureDesc* get() const { 
+      return m_pRootSignature;
+  }
+  DxilVersionedRootSignatureDesc* get_mutable() { 
+      return const_cast<DxilVersionedRootSignatureDesc *>(m_pRootSignature);
```

Comment:
I think it would be preferable for the private member variable to be non-const, then avoid the const_cast here.

Reply:
Your call, of course. The const_cast either has to be here or in get_address_of and the constructor. That is, if we want to be able to use one of these wrappers with DeserializeRootSignature.